bee-vote/src/fpc.rs

```diff
@@ -544,13 +540,44 @@ where
         all_queried_opinions.write().await.push(queried_opinions);
     }
 
+    /// Calculate liked percentage for each vote context.
+    async fn calculate_liked_percentages(&self, vote_map: HashMap<VoteObject, Opinions>) {
+        let mut contexts_guard = self.contexts.write().await;
+
+        for (object, votes) in vote_map.iter() {
+            let mut liked_sum = 0.0;
+            let mut voted_count = votes.len() as u32;
+
+            for vote in votes.iter() {
+                match vote {
+                    Opinion::Unknown => voted_count -= 1,
+                    Opinion::Like => liked_sum += 1.0,
+                    _ => {}
+                }
+            }
+
+            // This should never happen – there should always be a context for a given vote.
+            contexts_guard.get_mut(object).unwrap().round_completed();
+
+            // Make sure enough opinions were received for the round to be considered valid.
+            if voted_count < self.min_opinions_received {
+                continue;
+            }
+
+            contexts_guard
+                .get_mut(object)
+                .unwrap()
+                .set_liked(liked_sum / voted_count as f64);
```

Comment:
could it happen that all opinions are unknown and this turns out to be a division by zero?

Reply:
Nice catch, yes that's possible. I'll add a check to make sure `min_opinions_received` cannot be set to 0, therefore if `voted_count` is 0 this will be skipped.